File: task-03/C++/nPrime.cpp
#include <iostream>
#include <vector>
#include <cmath>

bool isPrime(int num) {
    if (num < 2) return false;
    for (int i = 2; i <= sqrt(num); i++) {
        if (num % i == 0) return false;
    }
    return true;
}

std::vector<int> findPrimes(int n) {
    std::vector<int> primes;
    for (int i = 2; i <= n; i++) {
        if (isPrime(i)) {
            primes.push_back(i);
        }
    }
    return primes;
}

int main() {
    int n;
    std::cout << "Enter a number: ";
    std::cin >> n;
    std::vector<int> result = findPrimes(n);
    std::cout << "Prime numbers up to " << n << " are: ";
    for (int prime : result) {
        std::cout << prime << " ";
    }
    std::cout << std::endl;
    return 0;
}

File: task-03/Rust/nPrime.rs
use std::io;

fn is_prime(num: u32) -> bool {
    if num < 2 {
        return false;
    }
    for i in 2..=((num as f64).sqrt() as u32) {
        if num % i == 0 {
            return false;
        }
    }
    true
}

fn find_primes(n: u32) -> Vec<u32> {
    let mut primes = Vec::new();
    for i in 2..=n {
        if is_prime(i) {
            primes.push(i);
        }
    }
    primes
}

fn main() {
    println!("Enter a number: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let n: u32 = input.trim().parse().expect("Please enter a valid number");

    let result = find_primes(n);
    println!("Prime numbers up to {} are: {:?}", n, result);
}
